sponsorship_switzerland/models/contracts.py

```diff
@@ -268,11 +270,19 @@ def _check_sponsorship_is_valid(self):
         Called at contract validation to ensure we can validate the
         sponsorship.
         """
-        if self.mapped("partner_id.activity_ids") or self.mapped(
-                "correspondent_id.activity_ids"):
+        check_duplicate_activity_id = self.env.ref("partner_compassion.activity_check_duplicates").id
+        if self.mapped("partner_id.activity_ids")\
+                .filtered(lambda l: l.activity_type_id.id == check_duplicate_activity_id)\
+                or self.mapped("correspondent_id.activity_ids")\
+                .filtered(lambda l: l.activity_type_id.id == check_duplicate_activity_id):
             raise UserError(
                 _("Please verify the partner before validating the sponsorship")
             )
+        # if self.mapped("partner_id.activity_ids") or self.mapped(
+        #         "correspondent_id.activity_ids"):
+        #     raise UserError(
+        #         _("Please verify the partner before validating the sponsorship")
+        #     )
```

Comment:
I prefer that we remove old code instead of commenting it. Git already enables us to go back in history if needed and thus we keep our clode clean.